GroceryApplication/app/src/main/java/com/company/arodrigues/groceryapplication/MainActivity.java

```diff
@@ -0,0 +1,98 @@
+package com.company.arodrigues.groceryapplication;
+
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ListView;
+
+import java.util.ArrayList;
+
+public class MainActivity extends AppCompatActivity {
+    public ArrayList<String> groceryList = new ArrayList<>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.fragment_main);
+        final EditText groceryItem = (EditText) findViewById(R.id.edittext);
+        Button myButton = (Button) findViewById(R.id.button);
+        final ListView listview = (ListView) findViewById(R.id.listview);
+        final ArrayAdapter<String> itemsAdapter = new ArrayAdapter(MainActivity.this,R.layout.grocery_list_layout, groceryList);
+        listview.setAdapter(itemsAdapter);
+        myButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (groceryItem.getText() != null) {
+                    groceryList.add(groceryItem.getText().toString());
+                    itemsAdapter.notifyDataSetChanged();
+                    groceryItem.setText("");
+                } else {
+                }
+            }
+        });
+        listview.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            public void onItemClick(AdapterView<?> parent, final View view, final int position, long id) {
+                final String grocItem = (String) parent.getItemAtPosition(position);
+                final EditText changeText = new EditText(MainActivity.this);
+                final AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
+                builder.setTitle(R.string.message);
+                builder.setView(changeText);
+                builder.setPositiveButton(R.string.cancel, new DialogInterface.OnClickListener() {
```

Comment:
Generally your positive button should be the one that is doing action and the negative button should be the one that is cancelling the dialog.